### PR TITLE
cleanup easy issues WD-06

### PR DESCRIPTION
**Pull request by equalsJeffH**
this fixes the easier of the "easy" issues: #285 #326 #327 #329 #351 #392 #490 #494, please review.

***
[Preview](https://s3.amazonaws.com/pr-preview/w3c/webauthn/jeffh-cleanup-easy-issues-wd-06.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/webauthn/d8b103a...fe2d92e.html)

### CODE REVIEW

**Comment by equalsJeffH** [comment]
This is an easy-peasy one (i hope) -- pls review :)

File: index.bs [hunk 1]
```diff
@@ -1219,7 +1219,7 @@ following Web IDL.
     dictionary CollectedClientData {
         required DOMString           challenge;
         required DOMString           origin;
-        required DOMString           hashAlg;
+        required DOMString           hashAlgorithm;
```
Comment (jyasskin) [inline index.bs]: IIUC, #326 and #351 say that the CollectedClientData and PublicKeyCredentialParameters fields should have consistent names. Here you've switched which one uses the short form, but they're still inconsistent. Maybe unify on "alg"?

File: index.bs [hunk 2]
```diff
@@ -973,24 +973,24 @@ optionally evidence of [=user consent=] to a specific transaction.
 
     The <dfn>type</dfn> member specifies the type of credential to be created.
 
-    The <dfn>algorithm</dfn> member specifies the cryptographic signature algorithm with which the newly generated credential
+    The <dfn>alg</dfn> member specifies the cryptographic signature algorithm with which the newly generated credential
```
Comment (equalsJeffH) [inline index.bs]: done in my local copy.

File: index.bs (hunk 1 above)
Comment (equalsJeffH) [inline index.bs]: well, AFAICT "alg" is only used in the spec IDL or prose in two places:
```
> grep -iE "alg[;\s}\.]+" index.bs 
        <code>|current|.{{PublicKeyCredentialParameters/alg}}</code> and |op| set to `"generateKey"`. If an error occurs during
        required AlgorithmIdentifier          alg;
```

..(tho my egrep fu is so-so) and so I'm inclined to leave this as-is with the vast majority of the use of [Aa]lgrorithm (85 instances) being the spelled-out form. 

that ok?

Comment (jyasskin) [inline index.bs]: I don't feel strongly. @jovasco @bfgeek, do you?

**Comment by equalsJeffH** [comment]
thanks for your review @jyasskin :)

File: index.bs (hunk 1 above)
Comment (jovasco) [inline index.bs]: I'm not up to date enough to understand which instance this is but please don't use multiple strings for the same thing towards the authenticator. It's all data we have to store and we win nothing. I rather use my <256k of flash storage for code and key storage than pretty member names.

File: index.bs [hunk 3]
```diff
@@ -482,7 +482,7 @@ the {{CredentialCreationOptions}} dictionary as follows:
 
 <pre class="idl">
     partial dictionary CredentialCreationOptions {
-        MakeCredentialOptions? publicKey;
+        MakeCredentialOptions      publicKey;
```
Comment (battre) [inline index.bs]: See my comment on the bug, I think this needs to stay nullable.

File: index.bs [hunk 4]
```diff
@@ -493,7 +493,7 @@ To support obtaining assertions via {{CredentialsContainer/get()|navigator.crede
 
 <pre class="idl">
     partial dictionary CredentialRequestOptions {
-        PublicKeyCredentialRequestOptions? publicKey;
+        PublicKeyCredentialRequestOptions      publicKey;
```
Comment (battre) [inline index.bs]: This needs to stay nullable in my opinion as well.

File: index.bs (hunk 3 above)
Comment (battre) [inline index.bs]: Taking this back after being educated by @jyasskin in issue #490 .

File: index.bs (hunk 4 above)
Comment (battre) [inline index.bs]: Taking this back as well.

File: index.bs (hunk 1 above)
Comment (equalsJeffH) [inline index.bs]: AFAICT, the short form, "alg", is used in all the data structures handled by authenticators. 

improved egrep for "alg":
```
> grep -niE "alg[:;\s}\.| ]+" index.bs
579:    1. Let |normalizedAlgorithm| be the result of [=normalizing an algorithm=] [[!WebCryptoAPI]], with |alg| set to
580:        <code>|current|.{{PublicKeyCredentialParameters/alg}}</code> and |op| set to `"generateKey"`. If an error occurs during
989:        required AlgorithmIdentifier          alg;
1711:            ; All public key formats must include an alg name
1712:            pubKeyTemplate = { alg: text }
1716:            rsaPubKey = { alg: rsaAlgName, n: biguint, e: uint }
1720:            eccPubKey = { alg: eccAlgName, x: biguint, y: biguint }
2075:                           alg: rsaAlgName / eccAlgName,
2080:                           alg: "ED256" / "ED512",
2114:    mechanism. It sets |x5c| to the certificate chain of the attestation public key and |alg| to the algorithm of the
2119:    through an authenticator-specific mechanism (see [[!FIDOEcdaaAlgorithm]]). It sets |alg| to the algorithm of the
2123:    and signing the result using the credential private key. It sets |alg| to the algorithm of the credential private key, and
2134:        attestation public key in |x5c| with the algorithm specified in |alg|.
2146:    - Validate that |alg| matches the algorithm of the credential private key in |authenticatorData|.
2148:        credential public key with |alg|.
2200:                            alg: rsaAlgName / eccAlgName,
2204:                            alg:  "ED256" / "ED512",
2273:        algorithm specified in |alg|.
2932:        ...                                         -- all public key alg etc.
2982:        ...                                         -- all public key alg etc.
3056:        ...                    -- all public key alg etc.
```

**Comment by equalsJeffH** [comment]
@jyasskin cautioned:
> But do give folks a chance to reply about how to resolve the alg/algorithm inconsistency.

Of course.  I  think we are good wrt "alg" and "algorithm" at this point (we use the latter preponderantly).  I note that web crypto uses "alg" in various of its var names -- it's a common abbreviation. 

thanks everyone for your reviews. 

Merge this?

**Comment by AngeloKai** [comment]
I am hoping @jcjones, @leshi, or @kpaulh can take a look at this before we merge.

**Comment by kpaulh** [comment]
These changes make me happy. For my part, I'm fine with the current usage of alg vs algorithm.

**Comment by equalsJeffH** [comment]
thx @selfissued 
> can you also please file a CTAP bug listing the changes made in WebAuthn that need to be tracked in CTAP? This at least includes "algorithm" -> "alg". Are there any others?

will do. 

> Are there any others?

I do not think so. 

NOTE:  In working with @selfissued to review this PR, I noticed some rough edges that I neglected to polish when doing PR #427, so I submitted issue #494, fixed that just now, and just  those latter changes can be quickly reviewed here: https://github.com/w3c/webauthn/pull/489/commits/fe2d92e4c6978d0ecd3e6b355b9b3185789b2a9d

thanks!

**Comment by equalsJeffH** [comment]
@selfissued requested:
> can you also please file a CTAP bug listing the changes made in WebAuthn that need to be tracked in CTAP? This at least includes "algorithm" -> "alg". 

done see fido-alliance/fido-2-specs#252

File: index.bs (hunk 1 above)
Comment (jcjones) [inline index.bs]: Since this doesn't go on the wire, I'd prefer the spelled-out form. Not willing to die on the hill though.

**Comment by AngeloKai** [comment]
I was talking with @selfissued about the spelling out thing. It seemed like a good idea to keep all the parameter names in the CTAP spec as numbers and create a table/map in the spec about how the numbers map to the names. This way we don't have to keep struggling with keeping the names short for slow transmission protocols, such as BLEs. (Ideas credit to @selfissued and AlexR) Should we perhaps wait a little bit on the merge until this is mauled over in the CTAP space? I am not the expert there so please let me know what you think.

**Comment by jyasskin** [comment]
If the only reason to use an abbreviated name here is to reduce the encoded CBOR size for protocols like CTAP, and CTAP encodes the name as an integer rather than the Javascript field name, then it seems like we shouldn't abbreviate the name in Javascript. So, #351 would just be fixed by unifying on "algorithm".

**Comment by jyasskin** [comment]
I wouldn't wait on merging the rest of this change, just remove the changes to `PublicKeyCredentialParameters.algorithm` from it before merging.

**Comment by equalsJeffH** [comment]
@jyasskin mused:
> ...CTAP encodes the name as an integer rather than the Javascript field name,

Well, at this time, that is an if-CTAP-is-modified-to-do-so kind of statement.   So I suggest we merge this as-is and if CTAP goes that direction we can then "just remove the changes to PublicKeyCredentialParameters.algorithm".

Ok?

**Comment by selfissued** [comment]
Please proceed with the merge today.  If the easy stuff is taking this long, it makes me fear how long the hard stuff may take us to do.

**Comment by jyasskin** [comment]
I'm happy merging either way. I prefer to avoid making two changes to these names, but it's not a huge deal to me.

**Comment by jovasco** [comment]
@AngeloKai 
I am not against changing the fields names to numbers but this has been discussed in the past and it was rejected back then.

**Pull request merged**